spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarProducerFactory.java

```diff
@@ -119,4 +115,27 @@ public Map<String, Object> getProducerConfig() {
 		return this.producerConfig;
 	}
 
+	private static <T> void loadConf(ProducerBuilder<T> producerBuilder, Map<String, Object> config) {
+		producerBuilder.loadConf(config);
+
+		// Workaround because encryptionKeys are not loaded by loadConf and can't be
```

Comment:
I believe the part of the comment that belongs in here now is the same one below for the other fields. The fact about the builder not offering "replace" is related to us overriding the entry in the config map (in the calling method).

I think we should put [these lines](https://github.com/spring-projects-experimental/spring-pulsar/pull/189/files#diff-90f0737a6ca3b93ae30e9446722bf7d1d5010723b3fa1df1a4ff99f06dbd4c00R123-R127) under [line 129](https://github.com/spring-projects-experimental/spring-pulsar/pull/189/files#diff-90f0737a6ca3b93ae30e9446722bf7d1d5010723b3fa1df1a4ff99f06dbd4c00R129) and move this comment to the [calling method](https://github.com/spring-projects-experimental/spring-pulsar/pull/189/files#diff-90f0737a6ca3b93ae30e9446722bf7d1d5010723b3fa1df1a4ff99f06dbd4c00R101) w/ slight adjustment to:


```java
// Replace default keys - workaround as they can't be replaced through the builder
if (encryptionKeys != null) {
  config.put("encryptionKeys", encryptionKeys);
}
```

I'm totally not trying to be nit-picky or over-prescriptive but I just realized why I was getting tripped up on the updated code - as these 2 things have split across 2 methods now. I think it may help newcomers later on understand what part of the code is for what caveat.

Reply:
done